contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -333,11 +360,24 @@ def _transcribe(item, parents, editRate, masterMobs):
                 otio.opentime.RationalTime(media_start, editRate),
                 otio.opentime.RationalTime(media_length, editRate)
             )
-            # copy the metadata from the master into the media_reference
-            media.metadata["AAF"] = mastermob_child.metadata.get("AAF", {})
+
+            # Copy the metadata from the master into the media_reference
+            clipMetadata = mastermob_child.metadata.get("AAF", {})
+
+            # If the composition was holding UserComments and the current masterMob has
+            # no UserComments, use the ones from the CompositionMob. But if the
+            # masterMob has any, prefer them over the compositionMob, since the
+            # masterMob is the ultimate authority for a source clip.
+            if compositionUserMetadata:
+                if "UserComments" not in clipMetadata.keys():
+                    clipMetadata['UserComments'] = compositionUserMetadata
+
+            media.metadata["AAF"] = clipMetadata
```

Comment:
I'm slightly confused by the variable name `clipMetadata` since it is going into the media reference. Did it come from the clip?

Reply:
Hm, the metadata will be assigned to the clip as well as the media reference. From my perspective it represents the metadata for a given clip.

Reply:
does it make sense to apply it to the media reference then?  Or is the media reference a reference to the AAF file?

Reply:
Ah, okay - this makes sense to be now.